src/wpool_process.erl

```diff
@@ -130,37 +155,55 @@ init({Name, Mod, InitArgs, LOptions}) ->
 %% @private
 -spec terminate(atom(), state()) -> term().
 terminate(Reason, State) ->
-    #state{mod = Mod,
+    #state{mod = #callback_cache{module = Mod},
            state = ModState,
            name = Name,
            options = Options} =
         State,
     ok = notify_queue_manager(worker_dead, Name, Options),
     wpool_process_callbacks:notify(handle_worker_death, Options, [Name, Reason]),
-    Mod:terminate(Reason, ModState).
+    case erlang:function_exported(Mod, terminate, 2) of
+        true ->
+            Mod:terminate(Reason, ModState);
+        _ ->
+            ok
+    end.
 
 %% @private
 -spec code_change(string(), state(), any()) -> {ok, state()} | {error, term()}.
-code_change(OldVsn, State, Extra) ->
-    case (State#state.mod):code_change(OldVsn, State#state.state, Extra) of
-        {ok, NewState} ->
-            {ok, State#state{state = NewState}};
-        Error ->
-            {error, Error}
+code_change(OldVsn, #state{mod = #callback_cache{module = Mod}} = State, Extra) ->
+    case erlang:function_exported(Mod, code_change, 3) of
+        true ->
+            case Mod:code_change(OldVsn, State#state.state, Extra) of
+                {ok, NewState} ->
+                    {ok, State#state{state = NewState}};
+                Error ->
+                    {error, Error}
```

Comment:
This was poorly implemented originally. If you don't mind, considering that the spec for `code_change/3` already contemplates returning errors as tuples, can you change it to…

```suggestion
                {error, Error} ->
                    {error, Error}
```
…or…
```suggestion
                Error ->
                    Error
```
…?

Reply:
Surely!